integration-test-groups/aws2/aws2-sqs-sns/src/test/java/org/apache/camel/quarkus/component/aws2/sqs/it/Aws2SqsSnsTest.java

```diff
@@ -40,28 +45,157 @@
 
     @Test
     public void sqs() {
-        final Config config = ConfigProvider.getConfig();
-        final String queueName = config.getValue("aws-sqs.queue-name", String.class);
+        final String queueName = getPredefinedQueueName();
 
-        String[] queues = RestAssured.get("/aws2-sqs-sns/sqs/queues")
+        final String[] queues = listQueues();
+        Assertions.assertTrue(Stream.of(queues).anyMatch(url -> url.contains(queueName)));
+
+        final String msg = sendSingleMessageToQueue(queueName);
+        awaitMessageFromQueue(msg, queueName);
+    }
+
+    private String[] listQueues() {
+        return RestAssured.get("/aws2-sqs-sns/sqs/queues")
                 .then()
                 .statusCode(200)
                 .extract()
                 .body().as(String[].class);
-        Assertions.assertTrue(Stream.of(queues).anyMatch(url -> url.contains(queueName)));
+    }
+
+    @Test
+    public void sqsDeleteMessage() {
+        final String qName = getPredefinedQueueName();
+        final String msg = sendSingleMessageToQueue(qName);
+        final String receipt = receiveReceiptOfMessageFromQueue(qName);
+        deleteMessageFromQueue(qName, receipt);
+        Assertions.assertNotEquals(receiveMessageFromQueue(qName), msg);
+    }
+
+    private String getPredefinedQueueName() {
+        return ConfigProvider.getConfig().getValue("aws-sqs.queue-name", String.class);
+    }
 
+    private String sendSingleMessageToQueue(String queueName) {
         final String msg = "sqs" + UUID.randomUUID().toString().replace("-", "");
         RestAssured.given()
                 .contentType(ContentType.TEXT)
                 .body(msg)
-                .post("/aws2-sqs-sns/sqs/send")
+                .post("/aws2-sqs-sns/sqs/send/" + queueName)
                 .then()
                 .statusCode(201);
+        return msg;
+    }
+
+    private String receiveReceiptOfMessageFromQueue(String queueName) {
+        return RestAssured.get("/aws2-sqs-sns/sqs/receive/receipt/" + queueName)
+                .then()
+                .statusCode(200)
+                .extract()
+                .body()
+                .asString();
+    }
+
+    private void deleteMessageFromQueue(String queueName, String receipt) {
+        RestAssured.delete("/aws2-sqs-sns/sqs/delete/message/" + queueName + "/" + receipt)
+                .then()
+                .statusCode(200);
+    }
+
+    @Test
+    public void sqsAutoCreateDelayedQueue() {
+        final String qName = "delayQueue";
+        final int delay = 10;
+        createDelayQueueAndVerifyExistence("delayQueue", 10);
+        final String msgSent = sendSingleMessageToQueue(qName);
+        Instant start = Instant.now();
+        awaitMessageFromQueue(msgSent, qName);
+        Assertions.assertTrue(Duration.between(start, Instant.now()).getSeconds() >= delay);
+        deleteQueue(qName);
```

Comment:
From the PR comment, could we say that deleteQueue is not working in camel at the moment ? And then should be fixed with camel 3.12 ? So, how could the test passes ?

Reply:
Yes, deleteQueue is currently not working and the test is failing. I'm not sure if we should wait with this PR for the changes in camel or drop deleteQueue and add it later. What do you suggest?

Reply:
In such situations, we tend to push obstacles from the way :) Like, marking the junit test as @  [Disabled](https://howtodoinjava.com/junit5/junit-5-disabled-test-example/#:~:text=JUnit%20%40Disabled%20annotation%20can%20be,reason%20this%20test%20is%20disabled.)("https: //camel-quarkus-issue-xyz"). And [create a camel-quarkus-issue-xyz](https://github.com/apache/camel-quarkus/issues/new) with title [CAMEL-3.X] report the fix from CAMEL-6054 into camel-quarkus (something like [this ticket](https://github.com/apache/camel-quarkus/issues/2479)).

Reply:
Thank you!